### PR TITLE
Raedar docker-compose.

### PR DESCRIPTION
**Pull request by fahdjamy**
**Type of change**

- [ ] Bug fix (fixes an issue)
- [x] Chore (small new feature added to the project)
- [ ] New feature (change adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

**Explain what this PR does?**

- fix docker-compose errors.
- orchestrate docker for the app, Redis, and Postgres.
- add shell scripts to install and uninstall for easier setup.
- fix bugs and add CompileDemon to add auto-update for changes when running in dev mode.

**Motivation? (required)**

- Enable Docker Compose to allow for container orchestration and to 

**Provide test plan? (required)**

- Pull the branch
- Make sure you have docker installed or run scripts/install.sh file
- Run `docker-compose build`
- Check out your local-host in postman and try to sign-up a user URL `http://localhost:8080/api/v1/signup`.

**Any background context you want to provide?**

Previously, a user had to set up Redis, Postgres and Go on their machine, right-now all they need is Docker installed on their machine and then run one single command for the entire setup of the back-end on their machine

**What are the relevant Trello/Pivotal/Jira/Github Issues?**

[link](https://trello.com/c/IhATBcmn/56-add-redis-to-the-project)
[link](https://trello.com/c/JlTw8NOI/51-docker-compose)

**Screenshots (if appropriate)**

N/A

**Questions:**

N/A

[1]: https://medium.com/@martinkonicek/what-is-a-test-plan-8bfc840ec171#.y9lcuqqi9

### CODE REVIEW

**Review by huxaiphaer — APPROVED** [review]
LGTM Seeka

File: backend/go.sum [hunk 1]
```diff
@@ -24,6 +24,7 @@ github.com/erikstmartin/go-testdb v0.0.0-20160219214506-8d10e4a1bae5 h1:Yzb9+7DP
 github.com/erikstmartin/go-testdb v0.0.0-20160219214506-8d10e4a1bae5/go.mod h1:a2zkGnVExMxdzMo3M0Hi/3sEU+cWnZpSni0O6/Yb/P0=
 github.com/fatih/color v1.8.0 h1:5bzFgL+oy7JITMTxUPJ00n7VxmYd/PdMp5mHFX40/RY=
 github.com/fatih/color v1.8.0/go.mod h1:3l45GVGkyrnYNl9HoIjnp2NnNWvh6hLAqD8yTfGjnw8=
+github.com/fsnotify/fsnotify v1.4.7 h1:IXs+QLmnXW2CcXuY+8Mzv/fWEsPGWxqefPtCP5CnV9I=
```
Comment (huxaiphaer) [inline backend/go.sum:27]: Are you downloading all these files ? Nga nyinji

Comment (fahdjamy) [reply backend/go.sum:27]: Yes, sir, I am. think of the go.sum file as package-lock.json or yarn-lock.json and go.mod as package.json Sebbo.

Comment (huxaiphaer) [reply backend/go.sum:27]: Cool

**Pull request merged**